src/rpc/net.cpp

```diff
@@ -1014,52 +1015,105 @@ static RPCHelpMan sendmsgtopeer()
-                          obj.pushKV("total", node.addrman->Size());
-                          ret.pushKV("all_networks", obj);
-                          return ret;
-                      },
+        "\nProvides information about the node's address manager by returning the number of "
+        "addresses in the `new` and `tried` tables and their sum for all networks.\n"
+        "When setting the verbose flag, the address manager entries for the new "
+        "and tried table are also returned.\n"
+        "This RPC is for testing only.\n",
+        {
+            {"verbose", RPCArg::Type::BOOL, RPCArg::Default{false}, "True for a list of address manager entries by table, false for address counts only"},
+        },
+        {
+            RPCResult{"for verbose = false", RPCResult::Type::OBJ_DYN, "", "json object with network type as keys",
+                {
+                    {RPCResult::Type::OBJ, "network", "the network (" + Join(GetNetworkNames(), ", ") + ", all_networks )", {
+                        {RPCResult::Type::NUM, "new", "number of addresses in the new table, which represent potential peers the node has discovered but hasn't yet successfully connected to."},
+                        {RPCResult::Type::NUM, "tried", "number of addresses in the tried table, which represent peers the node has successfully connected to in the past."},
+                        {RPCResult::Type::NUM, "total", "total number of addresses in both new/tried tables"},
+                    }},
+                },
+            },
+            RPCResult{"for verbose = true", RPCResult::Type::OBJ_DYN, "", "json object with network type, new_table and tried_table as keys",
+                {
+                    {RPCResult::Type::ELISION, "", "The same output as verbose = false"},
+                    {
+                        RPCResult::Type::ARR, "table", "list of addresses in the address manager table ( new_table, tried_table )",
+                        {
+                            {RPCResult::Type::OBJ, "", "an address manager table entry", {
+                                {RPCResult::Type::STR, "address", "the address"},
+                                {RPCResult::Type::NUM, "services", "the services the node might support"},
+                                {RPCResult::Type::NUM, "bucket", "the address manager bucket the address is placed in"},
+                                {RPCResult::Type::NUM, "position", "the bucket position the address is placed in"},
+                                {RPCResult::Type::STR, "source", "the address that relayed the address to us"},
+                            }},
+                        },
+                    },
+                },
+            },
+        },
+        RPCExamples{
+            HelpExampleCli("getaddrmaninfo", "")
+            + HelpExampleCli("getaddrmaninfo", "true")
+            + HelpExampleRpc("getaddrmaninfo", "")},
```

Comment:
In 56adcf42805b3007b66396754a9e5dcd0968d027: We could add the 'verbose' example for the RPC.
```suggestion
            + HelpExampleRpc("getaddrmaninfo", "")},
            + HelpExampleRpc("getaddrmaninfo", "true")}
```

Reply:
done!